modulemd/meson.build

```
@@ -381,35 +386,46 @@ endforeach
 
 
 # Fake test to ensure that all sources and headers are formatted properly
+clang_files = modulemd_srcs + \
+              modulemd_hdrs + \
+              modulemd_priv_hdrs + \
+              modulemd_validator_srcs + \
+              test_srcs + \
+              test_priv_hdrs \
+
 clang_args = [ '-i' ]
 test('clang_format', clang_format,
-     args : clang_args +
-            modulemd_srcs +
-            modulemd_hdrs +
-            modulemd_priv_hdrs +
-            modulemd_validator_srcs +
-            test_srcs +
-            test_priv_hdrs)
+     args : clang_args + clang_files)
+
+clang_tidy_args = [ '-p', meson.build_root(),
+                    '--checks=*,-llvm-header-guard,-bugprone-macro-parentheses',
```

```suggestion
                    '--checks=*,-llvm-header-guard,-bugprone-macro-parentheses,-misc-unused-parameters',
```
Upon doing some investigation, it looks like this change should help with the travis tests. Without it, `clang-tidy` is making some undesirable changes.

---

Would you mind giving me an example of such a case where an undesirable change is being made?

---

Exactly what seems to have caused the latest Fedora 29 travis run to fail, such as can be seen at the end of the raw log file: https://api.travis-ci.org/v3/job/596767803/log.txt

When I run the clang_tidy test locally without disabling the misc-unused-parameters check, I get dozens of undesirable changes along the lines of:

```
--- a/modulemd/modulemd-defaults-v1.c
+++ b/modulemd/modulemd-defaults-v1.c
@@ -461,7 +461,7 @@ modulemd_defaults_v1_get_default_profiles_for_stream_as_strv (
 static void
 modulemd_defaults_v1_get_property (GObject *object,
                                    guint prop_id,
-                                   GValue *value,
+                                   GValue * /*value*/,
                                    GParamSpec *pspec)
 {
   switch (prop_id)
```

---

So, it seems that clang-tidy 7.0.0 has a bug with that particular function. The 9.0.0 I have on my F31 system does not do that. I've modified the meson build to just require at least 9.0.0 for clang-tidy or else fall back to clang-format. That should make life more bearable. I'm going to cancel this PR and push a new branch (atop my #383 PR, since it fixes a bug that the clang-tidy stuff revealed).